src/Engine/Trace/Trace.cpp

```
@@ -447,56 +448,245 @@ Trace::GetPoints(TracePointVector &v, const Time min_time,
+  const unsigned denom = max_points - 1;
+  const unsigned need = (numer + denom - 1) / denom;
+  return std::max(stride, need);
+}
+
+void
+FilterTraceByBounds(const TracePointVector &in,
+                    TracePointVector &out,
+                    const TrailSpatialFilter &filter) noexcept
+{
+  out.clear();
+
```

Cppcheck (2.21.0)

[error] Signed integer overflow for expression '0-1'.

(integerOverflow)